pywhy_graphs/algorithms/pag.py

```diff
@@ -1176,3 +1187,190 @@ def pag_to_mag(graph):
         mag.add_edge(u, v, mag.directed_edge_name)
 
     return mag
+
+
+def legal_pag(G: PAG, L: Optional[set] = None, S: Optional[set] = None):
```

Comment:
This can also be turned into a private function since it just performing a subset of the `valid_pag` check

Maybe `_check_pag_definition` and then elaborate in the docstring what is checked.

Reply:
For some reason it took a lot of brainpower for me to come up with "legal" since I was already using "valid". But I will change it to what you suggested since that is much better.

Reply:
Oh I see the import does not work if this is not public? Okay maybe just rename to `check_pag_definition` and then we could just not document it in the public API. That would work too